### DISCUSSION TITLE
Two levels of achievement

### DISCUSSION THREAD

**Discussion by ke352802081770314** [iSoron/uhabits#1007] [Feature Requests]
I have some habits that I still mark when I only "sort of did" them but I would prefer to be aware of whether I did it properly so that in the timeline I can detect a deterioration (more often only sort-of-done) before failing altogether. Perhaps this should also affect the score, but even if not, tracking it will be helpful.

For example I created a daily habit "breakfast/eat fried tofu". My goal ideally is that I will have fried tofu for breakfast every morning. But if I ran short on time or got distracted or whatever, I still believe it is good for me to eat this dish later in the day, that's why I chose a less stringent wording. But I am still convinced that it will help me establish a robust habit more if I really have it as my first meal, right in the morning.
Or, "brush my teeth in the evening": if I fell into bed like a stone and wake up to go to the toilet during the night, loop habit tracker encourages me to brush my teeth then because I can still tick it off, and indeed it is still better than not doing it at all but not as good as having done it before going to bed. 

I am aware of the "skip" option but I consider this a different thing, because skip is for not doing a habit (due to circumstances preventing it) while inferiorly doing it is still doing it.

I am still a bit unsure myself how to label the two levels. One variant I first thought of would be:
* done (like now) represented by check mark
* done sub-optimally (working title) with some other symbol 
the other variant would be 
* done perfectly represented by a star ✶
* done (like now) represented by check mark

While originally I was thinking of the first variant, after mulling over it a bit, I think the second one would be better. It gives positive feedback for a flawless execution of the habit, and it doesn't create the notion that half-done might be good enough by watering down what done means, which may be bad for habit forming to people who haven't considered this slippery path before.

I am aware that this is yet another state for non-numeric habits which makes the cycle-through input method less convenient, but this has been challenged before and I don't consider it optimal even now. 
For numeric habits, "perfect" could be represented by a more ambitious goal but that only makes a difference and is worth considering if it influences scoring because the UI doesn't have space for symbolic feedback when it shows the number.

To prevent disruption to existing users, there should be an option enabling this feature, and like I mentioned with "skip" I would raise the question whether it should be a global option or one per habit. If it affects scoring, this would speak for a per-habit option to make it less disruptive.

**Comment by hiqua** [comment]
I think relevant keywords here are "elastic habit". Indeed that would be interesting, but in terms of UX/UI that makes everything more complicated.

**Comment by VA-MMXXII** [comment]
This would be useful to track intensity or cumulative things. If you just barely meet your exercise goal for the day, you get the checkmark, but if you really knocked it out of the park such that you did several days worth of exercise you might get an extra bold checkmark or one with an additional outline or highlight. And in the stats it might somehow also count for 2 or more days worth of exercise.

Couldn't an extra long press of the field trigger a popup where you could indicate intensity (2X, 3X, or whatever)?

**Comment by fejfo** [comment]
You could also use two habits, one for the strong version and one for the weak one.

**Comment by herrbrixter** [comment]
Duplicate of #1267
